nodes/explorer/src/api_backend/store.rs

```diff
@@ -44,3 +44,80 @@ where
         .collect();
     make_request_and_return_response!(futures::future::try_join_all(results))
 }
+
+#[derive(Deserialize)]
+pub(crate) struct BlocksQueryParams {
+    from: BlockId,
+    to: Option<BlockId>,
+}
+
+pub(crate) async fn blocks<Tx, S>(
+    State(store): State<OverwatchHandle>,
+    Query(query): Query<BlocksQueryParams>,
+) -> Response
+where
+    Tx: Transaction
+        + Clone
+        + Debug
+        + Eq
+        + Hash
+        + Serialize
+        + DeserializeOwned
+        + Send
+        + Sync
+        + 'static,
+    <Tx as Transaction>::Hash: std::cmp::Ord + Debug + Send + Sync + 'static,
+    S: StorageSerde + Send + Sync + 'static,
+{
+    let BlocksQueryParams { from, to } = query;
+    // get the from block
+    let from = match storage::block_req::<S, Tx>(&store, from).await {
+        Ok(from) => match from {
+            Some(from) => from,
+            None => {
+                return ::axum::response::IntoResponse::into_response((
+                    ::axum::http::StatusCode::NOT_FOUND,
+                    "from block not found",
+                ))
+            }
+        },
+        Err(e) => {
+            return ::axum::response::IntoResponse::into_response((
+                ::axum::http::StatusCode::INTERNAL_SERVER_ERROR,
+                e.to_string(),
+            ))
+        }
+    };
+
+    let mut cur = Some(from.header().parent());
+    let mut blocks = Vec::new();
+    while let Some(id) = cur {
+        if let Some(to) = to {
```

Comment:
I'd imagine by providing `to` parameter, user could verify that the block range is a valid range - meaning, block between `from` and `to` are a verified sequence. What if user provides invalid `to`? In this case the endpoint would return everything, including genesis. Shouldn't this either confirm the validity, or return blocks `from` to the `depth` provided by the user?

Reply:
Yeah, good point! I will add a check for `to`. For the `depth`, I think we can return an error response if `to` is invalid, and having another API whose params are `from` and `depth` is better. Because if we add the `depth` in this API, what if the user provides both a valid `to` and also provides a `depth`, the final result should return `from` to `depth` or `from` to `to`, this will bring ambiguity. What is your opinion?

Reply:
@danielSanchezQ maybe you could confirm the purpose of `from` and `to` params, when they are both provided by the user and they both are `BlockId`?

> I will use Either<BlockId, usize> for to param. This should solve the ambiguity.

Maybe we could still separate those into two routes? Or one, but where the user would need to specify if he's using `BlockId` or `usize` (not for us, but for users own clarity):
- `/blocks?filter="blockid"&from="0x1..."&to="0x2..."`

What do you think?

Reply:
For me, separating into two routes is better. This will give users a simplified URL.

Reply:
I think using two separated routes is better indeed:
* From id / to id
* from id / to depth